cnf-certification-test/lifecycle/podrecreation/podrecreation.go

```diff
@@ -0,0 +1,146 @@
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+package podrecreation
+
+import (
+	"context"
+
+	"github.com/sirupsen/logrus"
+	"github.com/test-network-function/cnf-certification-test/internal/clientsholder"
+	"github.com/test-network-function/cnf-certification-test/pkg/provider"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	ReplicaSetString  = "ReplicaSet"
+	DeploymentString  = "Deployment"
+	StatefulsetString = "StatefulSet"
+	DaemonSetString   = "DaemonSet"
+)
+
+func CordonNode(name string) error {
+	clients := clientsholder.GetClientsHolder()
+	// Fetch node object
+	node, err := clients.Coreclient.Nodes().Get(context.TODO(), name, metav1.GetOptions{})
+
+	if err != nil {
+		return err
+	}
+
+	node.Spec.Unschedulable = true
+
+	// Update the node
+	_, err = clients.Coreclient.Nodes().Update(context.TODO(), node, metav1.UpdateOptions{})
+
+	return err
+}
+
+func UncordonNode(name string) error {
+	clients := clientsholder.GetClientsHolder()
+	// Fetch node object
+	node, err := clients.Coreclient.Nodes().Get(context.TODO(), name, metav1.GetOptions{})
+
+	if err != nil {
+		return err
+	}
+
+	node.Spec.Unschedulable = false
+
+	// Update the node
+	_, err = clients.Coreclient.Nodes().Update(context.TODO(), node, metav1.UpdateOptions{})
+
+	return err
+}
+
+func DeletePods(nodeName string) {
+	clients := clientsholder.GetClientsHolder()
+	pods, err := clients.Coreclient.Pods("").List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		logrus.Errorf("error getting list of pods err: %s", err)
+	}
+	for idx := range pods.Items {
+		for _, or := range pods.Items[idx].OwnerReferences {
+			if pods.Items[idx].Spec.NodeName == nodeName && or.Kind != DaemonSetString {
+				err = clients.Coreclient.Pods(pods.Items[idx].Namespace).Delete(context.TODO(), pods.Items[idx].Name, metav1.DeleteOptions{})
+				if err != nil {
+					logrus.Errorf("error deleting pod %s err: %v", provider.PodToString(&pods.Items[idx]), err)
+				}
+			}
+		}
+	}
+}
+
+func CountPods(nodeName string) (count int) {
+	count = 0
+	clients := clientsholder.GetClientsHolder()
+	pods, err := clients.Coreclient.Pods("").List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		logrus.Errorf("error getting list of pods err: %s", err)
+	}
+	for idx := range pods.Items {
+		for _, or := range pods.Items[idx].OwnerReferences {
+			if pods.Items[idx].Spec.NodeName == nodeName && or.Kind != DaemonSetString {
+				count++
+			}
+		}
+	}
+	return count
+}
+
+func GetDeploymentNodes(pods []*v1.Pod, dName string, ch *clientsholder.ClientsHolder) (nodes []string) {
+	for _, put := range pods {
+		deploymentFound := false
+		for _, or := range put.OwnerReferences {
+			if deploymentFound {
+				break
+			}
+			if or.Kind == ReplicaSetString {
+				r, err := ch.K8sClient.AppsV1().ReplicaSets(put.Namespace).Get(context.TODO(), or.Name, metav1.GetOptions{})
```

Comment:
Fixed. Reverted the TC to the original logic: Instead of testing deployment individually, now testing all the deployment together -> not need to identify a single owner anymore